examples/sdk/auth-social/src/pages/Register.tsx

```diff
@@ -0,0 +1,130 @@
+import { WebAuthnSigner } from '@dfns/sdk-browser'
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+import { CredentialResponse, GoogleLogin } from '@react-oauth/google';
+
+import '../globals.css'
+import { dfnsApi, setAuthToken } from '../api'
+import { useAppContext } from '../hooks/useAppContext';
+import { DfnsError, UserRegistrationChallenge } from '@dfns/sdk';
+
+export default function Register() {
+  const [loading, setLoading] = React.useState(false)
+  const [response, setResponse] = React.useState(undefined)
+  const [error, setError] = React.useState(undefined)
+  const { setAuthToken: setAuthTokenHook } = useAppContext()
+
+  const initRegistration = async (idToken: string) => {
+    try {
+      const challenge = await dfnsApi().auth.createSocialRegistrationChallenge({
+        body: {
+          socialLoginProviderKind: "Oidc",
+          idToken
+        },
+      })
+
+      return challenge
+    } catch (error: any) {
+      if (error instanceof DfnsError && error.httpStatus === 401) {
+        console.log("Already registered, using the logging route...")
+        return false
+      }
+      throw error
+    }
+  }
+
+  const finishRegistration = async (challenge: UserRegistrationChallenge) => {
+    // Webauthn flow
+    // Create the new webauthn credential using the challenge
+    const webauthn = new WebAuthnSigner()
+    const attestation = await webauthn.create(challenge)
+
+    // We set the temporary authentication token to complete the registration
+    setAuthToken(challenge.temporaryAuthenticationToken)
+
+    const result = await dfnsApi().auth.registerEndUser({
```

Comment:
nit: one suggestion is to make token to `dfnsApi` creation more explicit by making it an argument to the function call instead `dfnsApi(tempToken).auth`, rather than hiding it in the local storage helper.

the auth system can be confusing. I like to have examples to be as clear as possible that you are create this here with this parameter, even if it's not as nicely refactored as you would in a real application.